# Week 1 Assignment 1 - 'remove let'

SangYoonLee1231 opened the pull request:
### 첫 인사
- <strong>월요일 아침</strong>부터 학습을 진행했으나, 부족한 기본기로 인한 느린 진도로 인해 이제서야 PR을 보내게 되었습니다..  
  (PR도 이번이 처음입니다.)

- 비록 많이 늦었지만 잘 부탁드립니다.

<br/>

### 과제 수행 방식
- 변수를 재할당해선 안 된다고 하였기에, <strong>매개변수를 이용하여 값을 넘겨주는 방식</strong>으로 과제를 수행했습니다.

<br/>

SangYoonLee1231 committed:
# Week 1 Assignment 1 - 'remove let'

- 월요일 아침부터 학습을 진행했으나, 부족한 기본기로 인한 느린 진도로 인해 이제서야 PR을 보내게 되었습니다.. (PR도 이번이 처음입니다.)

- 변수를 재할당해선 안 된다고 하였기에, 매개변수를 이용하여 값을 넘겨주는 방식으로 과제를 수행했습니다.

daadaadaah commented on src/index.js:
```diff
@@ -1,2 +1,56 @@
+
+//
+
+const count = 0;
```
count 변수가 56번째 줄에서만 사용해서 따로 변수화가 필요 없을 거 같아요!ㅎㅎ

daadaadaah commented on src/index.js:
```diff
@@ -1,2 +1,56 @@
+
+const count = 0;
+
+function handleClick(count) {
```
render 함수에도 count 라는 이름을 사용하므로, 다른 네이밍으로 해주는게 좋을 것 같아요!ㅎㅎ

daadaadaah reviewed (changes requested):
늦어지만 이렇게 PR 날려주신 점 너무 잘하셨어요!👍
push 전에 lint와 test 한번 체크해주시면 좋을 것 같아요!ㅎㅎ
지금 test 에서 에러나고 있어요!

SangYoonLee1231 committed:
First Fix

- count 변수 선언을 없애고, 대신 render 함수의 매개변수에 초기값을 할당했습니다.

- handleClick 함수가 필요 없을 것 같아 삭제하고, 대신 button 요소가 onClick시 호출되는 함수를 handleClick 함수가 아닌 render 함수로 수정했습니다.

SangYoonLee1231 commented:
> 늦어지만 이렇게 PR 날려주신 점 너무 잘하셨어요!👍 push 전에 lint와 test 한번 체크해주시면 좋을 것 같아요!ㅎㅎ 지금 test 에서 에러나고 있어요!

저런.. 깜빡했군요. 죄송합니다. 리뷰해주신 내용 반영하여 다시 수정했습니다!

daadaadaah reviewed (changes requested):
https://github.com/CodeSoom/react-week1-assignment-1/pull/193/commits/c2dc6c757ce9acb7bdbd8cc35988b34581a7968c 수정 전 코드인
handleClick 함수를 따로 빼서 사용하시는게 나중에 유지보수하기에 더 괜찮을 것 같아요!

SangYoonLee1231 committed:
Second Fix

- 리뷰를 참고하여 유지 보수가 쉬워지도록 handleClick 함수를 따로 빼서 구현했습니다!

- 해설 강의를 참고하여 render 함수의 매개변수 안에서 count를 0으로 초기화 하는 대신, render 함수를 처음 호출 할 때(line 54) 정수 0을 넘겨주는 방식으로 초기화를 진행했습니다.

SangYoonLee1231 commented on src/index.js:
```diff
@@ -47,4 +51,4 @@ function render(count = 0) {
 }
 
-render();
+render(0);
```
혹시 바보 같은 질문 하나만 여쭤봐도 될까요..? 🤔

count 변수를 초기화 할 때,  수정 전 코드처럼 <strong>render 함수의 매개변수에서 초기화를 진행하는 방식</strong>과, 수정 후 코드처럼 <strong>render 함수 첫 호출 시 0을 넘겨주어 초기화를 진행하는 방식</strong> 중 어떤 코드가 더 보기 좋은 (가독성이 높은) 코드라고 말할 수 있을까요?
아님 그저 스타일의 차이라고 보면 될까요?

daadaadaah replied:
초기값이 0으로 지정되어 있다면, 저는 수정 전 코드가 더 좋은 것 같아요!ㅎㅎ
왜냐하면, 함수가 선언된 부분만 보고서 초기값을 알 수 있기 때문에, 좀더 함수를 명확히 알 수 있어서 더 좋은 것 같습니다!ㅎㅎ

SangYoonLee1231 committed:
npm test 결과 반영하여 오타 수정

- npm test를 뒤늦게 돌려보고 오타를 수정했습니다!
- 'Click Me!' → 'Click me!'

daadaadaah reviewed (approved):
수고하셨습니다👏
